pac4j-core/src/main/java/org/pac4j/core/profile/service/InMemoryProfileService.java

```diff
@@ -0,0 +1,101 @@
+package org.pac4j.core.profile.service;
+
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.exception.TechnicalException;
+import org.pac4j.core.profile.CommonProfile;
+import org.pac4j.core.profile.definition.CommonProfileDefinition;
+import org.pac4j.core.util.CommonHelper;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * In-memory profile service.
+ *
+ * @author Elie Roux
+ * @since 2.0.0
+ */
+
+public class InMemoryProfileService<U extends CommonProfile> extends AbstractProfileService<U>  {
+
+	public Map<String,Map<String,Object>> profiles;
+	//public AuthenticatorProfileCreator<UsernamePasswordCredentials, U> creator;
+	public Class<U> typeArgumentClass;
+
+	public InMemoryProfileService(Class<U> typeArgumentClass) {
+		this(new HashMap<String,Map<String,Object>>(), typeArgumentClass);
```

Comment:
Or even better, a `Function<Object[], U>` which is exactly what the `CommonProfileDefinition` is expecting…